### ISSUE TITLE
Updating examples to work with .net 8

### ISSUE THREAD

**Issue by rusty21** [asieradzk/RL_Matrix#14]
Any way you could update your samples to run on the latest nuget versions and also update them to use .net 8?

**Comment by asieradzk** [comment]
Yeah its coming. If you've been following repo/nugets you'll see im making changes every day. 

Which platform & workflow (local/remote) are you interested in? I'll push one quickly for you.

**Comment by asieradzk** [comment]
> I'm working on a windows machine using the local workflow.

https://github.com/asieradzk/CartPoleForTesting

Try this

Also clone main repo and build/run the dashboard so you can see whats going on.

**Comment by rusty21** [comment]
Looks like it compiles and run, after a small amount of time it seems like it implodes on it self, JSON errors show in the console window and the 4 experiment windows reset almost instantly. I'm assuming this is because you're still working out some bugs.

**Comment by asieradzk** [comment]
> Looks like it compiles and runs, after a small amount of time it seems like it implodes on it self, JSON errors show in the console window and the 4 experiment windows reset almost instantly. I'm assuming this is because you're still working out some bugs.

No thats odd. Should be fine... I am not even using any JSON. Can you show your errors?

**Comment by rusty21** [comment]
Heres a screenshot of what I experience. Just an endless loop.
![image](https://github.com/user-attachments/assets/44679c79-552a-4ef9-aa30-4006418143c9)

**Comment by asieradzk** [comment]
> Heres a screenshot of what I experience. Just an endless loop. 


This is because of exploding gradients:
https://machinelearningmastery.com/exploding-gradients-in-neural-networks/

You likely have one or combination of following things in your env:
-Too large learning rate
-using rnn
-small batch size

Also beware when using continuous env you must assign penalties (negative rewards) when action goes out of bounds. you can easily get NaNs there by clamping network output and doing some post-processing there without giving a signal that its approaching very large numbers.

**Comment by rusty21** [comment]
All I did was uncomment the cart pole example that you already had. I changed none of the parameters of the dqn agent. I admit I know next to nothing about how use and train these models. I know this is off topic from the original github issue post, but would you be willing to answer some other questions I have as a beginner?

**Comment by asieradzk** [comment]
> All I did was uncomment the cart pole example that you already had. I changed none of the parameters of the dqn agent. I admit I know next to nothing about how to use and train these models. I know this is off topic from the original github issue post, but would you be willing to answer some other questions I have as a beginner?

Yeah, deep reinforcement learning can be incredibly fragile. Sometimes you can end up with exploding gradients which lead to NaNs being output by network. Ask any questions you might have. :)
